scripts/lib/performance_tests.sh

```diff
@@ -0,0 +1,209 @@
+function check_for_timeout() {
+    if [[ $((SECONDS - $1)) -gt 10000 ]]; then
+        RUNNING_PERFORMANCE=false
+        on_error
+    fi
+}
+
+function run_performance_test_130_pods() {
+    echo "Running performance tests against cluster"
+    RUNNING_PERFORMANCE=true
+
+    DEPLOY_START=$SECONDS
+
+    SCALE_UP_DURATION_ARRAY=()
+    SCALE_DOWN_DURATION_ARRAY=()
+    while [ ${#SCALE_UP_DURATION_ARRAY[@]} -lt 3 ]
+    do
+        ITERATION_START=$SECONDS
+        $KUBECTL_PATH scale -f ./testdata/deploy-130-pods.yaml --replicas=130
+        sleep 20
+        while [[ ! $($KUBECTL_PATH get deploy | grep 130/130) ]]
+        do
+            sleep 1
+            echo "Scaling UP"
+            echo $($KUBECTL_PATH get deploy)
+            check_for_timeout $DEPLOY_START
+        done
+
+        SCALE_UP_DURATION_ARRAY+=( $((SECONDS - ITERATION_START)) )
+        MIDPOINT_START=$SECONDS
+        $KUBECTL_PATH scale -f ./testdata/deploy-130-pods.yaml --replicas=0
+        while [[ $($KUBECTL_PATH get pods) ]]
+        do
+            sleep 1
+            echo "Scaling DOWN"
+            echo $($KUBECTL_PATH get deploy)
+            check_for_timeout $DEPLOY_START
+        done
+        SCALE_DOWN_DURATION_ARRAY+=($((SECONDS - MIDPOINT_START)))
+    done
+
+    echo "Times to scale up:"
+    INDEX=0
+    while [ $INDEX -lt ${#SCALE_UP_DURATION_ARRAY[@]} ]
+    do
+        echo ${SCALE_UP_DURATION_ARRAY[$INDEX]}
+        INDEX=$((INDEX + 1))
+    done
+    echo ""
+    echo "Times to scale down:"
+    INDEX=0
+    while [ $INDEX -lt ${#SCALE_DOWN_DURATION_ARRAY[@]} ]
+    do
+        echo "${SCALE_DOWN_DURATION_ARRAY[$INDEX]} seconds"
+        INDEX=$((INDEX + 1))
+    done
+    echo ""
+    DEPLOY_DURATION=$((SECONDS - DEPLOY_START))
+
+    now="pod-130-Test#${TEST_ID}-$(date +"%m-%d-%Y-%T").csv"
+    echo $now
+
+    echo $(date +"%m-%d-%Y-%T") >> $now
+    echo $((SCALE_UP_DURATION_ARRAY[0])), $((SCALE_DOWN_DURATION_ARRAY[0])) >> $now
+    echo $((SCALE_UP_DURATION_ARRAY[1])), $((SCALE_DOWN_DURATION_ARRAY[1])) >> $now
+    echo $((SCALE_UP_DURATION_ARRAY[2])), $((SCALE_DOWN_DURATION_ARRAY[2])) >> $now
+
+    cat $now
+    aws s3 cp $now s3://cni-performance-test-data
```

Comment:
I think this bucket needs to be a configurable setting. And if it's not set, we should skip the upload. Something like
```bash
if [[ -n "${S3_PERF_TEST_BUCKET:-}" ]]; then
    aws s3 cp $filename "$S3_PERF_TEST_BUCKET"
else 
    echo "No S3 bucket name given, not uploading results"
fi
```

Reply:
Should we do the check for S3 bucket before the test?